### PR TITLE
 feat(files): download files with .peeriodownload suffix and then rename

### PR DESCRIPTION
**Pull request by dchest**
Accidentally included the other PR, will rebase after that one is merged.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

#### Relevant issue tracker links  

https://peerio.tpondemand.com/entity/7633

#### Linked PRs in other repos

Mobile rename for FileStream:
https://github.com/PeerioTechnologies/peerio-mobile/pull/68

#### Breaking changes  

None.

#### Manual testing instructions  

1. Download a file.
2. On desktop, watch the directory while it's being downloaded — see how it first appears as *filename*.peeriodownload.
3. Make sure the file is renamed to its original name (selected in Save dialog) after download finishes.

Do the same with a file that's already exists in the destination folder (when save dialog asks if you want to replace file). Make sure the file gets replaced.

#### Additional notes
... 

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Are there any bicycles invented that should be replaced with existing code or external libs?
* Are there unit and e2e tests added?

----
### Repository owner

- [ ] Breaking changes are applied correctly.
- [ ] At least one approved code review exists.
- [ ] Approved tester review exists and was done after last code changes.
- [ ] CI build and tests passed.
- [ ] This feature/bugfix is considered fully done and can get merged.
- [ ] PR name follows conventional changelog format.
- [ ] I'll remember to squash commits.

### CODE REVIEW

File: src/models/files/file-processor.js [hunk 1]
```diff
@@ -60,19 +60,25 @@ class FileProcessor {
         if (this.processFinished) return;
         this.processFinished = true;
         this.stopped = true; // bcs in case of error some calls might be scheduled
-        try {
-            this.stream.close();
-        } catch (e) {
```
Comment (anri-asaturov) [inline src/models/files/file-processor.js]: why did you decide to get rid of sychronous catcher?

Comment (dchest) [reply src/models/files/file-processor.js]: because `this.stream.close()` returns a promise.

Comment (dchest) [reply src/models/files/file-processor.js]: also `close` failure can lead to not writing the complete file.

Comment (anri-asaturov) [reply src/models/files/file-processor.js]: right, i just checked, this is a leftover after refactor

**Pull request merged**